### PR TITLE
Use guid

### PR DESCRIPTION
**Pull request by penk85**
Change id type to GUID, add source field for peer id.

### CODE REVIEW

File: packages/api/prisma/schema.prisma [hunk 1]
```diff
@@ -69,7 +69,7 @@ model ArticleRevision {
 }
 
 model Article {
-  id         String   @id @default(cuid())
+  id         String   @id @default(uuid())
```
Comment (tomaszdurka) [inline packages/api/prisma/schema.prisma:72]: I guess we can have default on postgresql but abitlity to set specific one. @Itrulia any thoughts on that? I actually thought about passing it from code.

File: packages/api/prisma/migrations/20221024073255_guids/migration.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-- AlterTable
+ALTER TABLE "articles" ADD COLUMN     "source" TEXT;
+
+-- AlterTable
+ALTER TABLE "authors" ADD COLUMN     "source" TEXT;
+
+-- AlterTable
+ALTER TABLE "images" ADD COLUMN     "origin" TEXT;
```
Comment (tomaszdurka) [inline packages/api/prisma/migrations/20221024073255_guids/migration.sql:8]: We definitely need one name for all.

What about `peerId`

Comment (tomaszdurka) [reply packages/api/prisma/migrations/20221024073255_guids/migration.sql:8]: `origin`, `producer` could work too.

Still we should make it connected with the `peers` table with relation and some on-cascades.

@Itrulia wdyt?

**Review by tomaszdurka — CHANGES_REQUESTED** [review]

File: packages/api/prisma/schema.prisma (hunk 1 above)
Comment (Itrulia) [reply packages/api/prisma/schema.prisma:72]: We probably should use the DB specific one ala https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#override-default-value-behavior-for-supported-types

Comment (Itrulia) [reply packages/api/prisma/schema.prisma:72]: My issue with this change is more so that old articles are not migrated.

Comment (tomaszdurka) [reply packages/api/prisma/schema.prisma:72]: Agree to Prisma internal. 

Yes, definitely we need migration in this PR too. 
As Karl suggested.
Making sure we have proper onUpdate cascades and setting to null. Hmm @Itrulia  won't this drop the relation when setting to `null` and then when re-adding it, it won't have the connections with other entities? I'm really not sure how this behaves. I would do some tests.

File: packages/api/prisma/migrations/20221024073255_guids/migration.sql (hunk 2 above)
Comment (Itrulia) [reply packages/api/prisma/migrations/20221024073255_guids/migration.sql:8]: peerId would clash with the id of the peer for now, would not use that until we have a guid on it.
I like origin

File: packages/api/prisma/schema.prisma (hunk 1 above)
Comment (Itrulia) [reply packages/api/prisma/schema.prisma:72]: Also checked: Cascade OnUpdate is default. So all we realistically have to do is set the id to the GUID function in the migration

**Review by Itrulia — APPROVED** [review]

File: packages/api/prisma/migrations/20221026074225_articles_images_authors_guid/migration.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
+-- AlterTable
+ALTER TABLE "articles" ADD COLUMN     "origin" TEXT,
+ALTER COLUMN "id" SET DEFAULT gen_random_uuid();
+
+-- AlterTable
```
Comment (Itrulia) [inline packages/api/prisma/migrations/20221026074225_articles_images_authors_guid/migration.sql:5]: Let's remove this for now and do this in a separate PR

File: packages/api/prisma/migrations/20221026074225_articles_images_authors_guid/migration.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ALTER TABLE "authors" ADD COLUMN     "origin" TEXT,
+ALTER COLUMN "id" SET DEFAULT gen_random_uuid();
+
+-- AlterTable
```
Comment (Itrulia) [inline packages/api/prisma/migrations/20221026074225_articles_images_authors_guid/migration.sql:9]: Let's remove this for now and do this in a separate PR

**Review by Itrulia — CHANGES_REQUESTED** [review]

**Comment by Itrulia** [comment]
SQL to migrate authors and images:

```sql
--Migrate authors to use GUIDs
ALTER TABLE "authors" ADD "oldId" TEXT NULL;
UPDATE "authors" SET "oldId" = "id";

UPDATE "authors" SET "id" = gen_random_uuid();

ALTER TABLE "articles.revisions.author" DROP CONSTRAINT "articles.revisions.author_authorId_fkey";
ALTER TABLE "articles.revisions.social-media-author" DROP CONSTRAINT "articles.revisions.social-media-author_authorId_fkey";
ALTER TABLE "authors.links" DROP CONSTRAINT "authors.links_authorId_fkey";

ALTER TABLE "authors" ALTER COLUMN "id" TYPE uuid USING "id"::uuid;

ALTER TABLE "articles.revisions.author" ALTER COLUMN "authorId" TYPE uuid USING "authorId"::uuid;
ALTER TABLE "articles.revisions.social-media-author" ALTER COLUMN "authorId" TYPE uuid USING "authorId"::uuid;
ALTER TABLE "authors.links" ALTER COLUMN "authorId" TYPE uuid USING "authorId"::uuid;

ALTER TABLE "articles.revisions.author" ADD CONSTRAINT "articles.revisions.author_authorId_fkey" FOREIGN KEY ("authorId") REFERENCES "authors"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "articles.revisions.social-media-author" ADD CONSTRAINT "articles.revisions.social-media-author_authorId_fkey" FOREIGN KEY ("authorId") REFERENCES "authors"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "authors.links" ADD CONSTRAINT "authors.links_authorId_fkey" FOREIGN KEY ("authorId") REFERENCES "authors"("id") ON DELETE CASCADE ON UPDATE CASCADE;


--Migrate images to use GUIDs
ALTER TABLE "peerProfiles" ADD CONSTRAINT "peerProfiles_callToActionImageID_fkey" FOREIGN KEY ("callToActionImageID") REFERENCES "images"("id") ON DELETE SET NULL ON UPDATE CASCADE;

ALTER TABLE "images" ADD "oldId" TEXT NULL;
UPDATE "images" SET "oldId" = "id";

UPDATE "images" SET "id" = gen_random_uuid();


ALTER TABLE "authors" DROP CONSTRAINT "authors_imageID_fkey";
ALTER TABLE "articles.revisions" DROP CONSTRAINT "articles.revisions_imageID_fkey";
ALTER TABLE "articles.revisions" DROP CONSTRAINT "articles.revisions_socialMediaImageID_fkey";
ALTER TABLE "pages.revision" DROP CONSTRAINT "pages.revision_imageID_fkey";
ALTER TABLE "pages.revision" DROP CONSTRAINT "pages.revision_socialMediaImageID_fkey";
ALTER TABLE "images.focal-point" DROP CONSTRAINT "images.focal-point_imageId_fkey";
ALTER TABLE "comments" DROP CONSTRAINT "comments_guestUserImageID_fkey";
ALTER TABLE "member.plans" DROP CONSTRAINT "member.plans_imageID_fkey";
ALTER TABLE "peerProfiles" DROP CONSTRAINT "peerProfiles_callToActionImageID_fkey";
ALTER TABLE "peerProfiles" DROP CONSTRAINT "peerProfiles_logoID_fkey";

ALTER TABLE "images" ALTER COLUMN "id" TYPE uuid USING "id"::uuid;

ALTER TABLE "authors" ALTER COLUMN "imageID" TYPE uuid USING "imageID"::uuid;
ALTER TABLE "articles.revisions" ALTER COLUMN "imageID" TYPE uuid USING "imageID"::uuid;
ALTER TABLE "articles.revisions" ALTER COLUMN "socialMediaImageID" TYPE uuid USING "socialMediaImageID"::uuid;
ALTER TABLE "pages.revision" ALTER COLUMN "imageID" TYPE uuid USING "imageID"::uuid;
ALTER TABLE "pages.revision" ALTER COLUMN "socialMediaImageID" TYPE uuid USING "socialMediaImageID"::uuid;
ALTER TABLE "images.focal-point" ALTER COLUMN "imageId" TYPE uuid USING "imageId"::uuid;
ALTER TABLE "comments" ALTER COLUMN "guestUserImageID" TYPE uuid USING "guestUserImageID"::uuid;
ALTER TABLE "member.plans" ALTER COLUMN "imageID" TYPE uuid USING "imageID"::uuid;
ALTER TABLE "peerProfiles" ALTER COLUMN "callToActionImageID" TYPE uuid USING "callToActionImageID"::uuid;
ALTER TABLE "peerProfiles" ALTER COLUMN "logoID" TYPE uuid USING "logoID"::uuid;

ALTER TABLE "authors" ADD CONSTRAINT "authors_imageID_fkey" FOREIGN KEY ("imageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "articles.revisions" ADD CONSTRAINT "articles.revisions_imageID_fkey" FOREIGN KEY ("imageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "articles.revisions" ADD CONSTRAINT "articles.revisions_socialMediaImageID_fkey" FOREIGN KEY ("socialMediaImageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "pages.revision" ADD CONSTRAINT "pages.revision_imageID_fkey" FOREIGN KEY ("imageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "pages.revision" ADD CONSTRAINT "pages.revision_socialMediaImageID_fkey" FOREIGN KEY ("socialMediaImageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "images.focal-point" ADD CONSTRAINT "images.focal-point_imageId_fkey" FOREIGN KEY ("imageId") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "comments" ADD CONSTRAINT "comments_guestUserImageID_fkey" FOREIGN KEY ("guestUserImageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "member.plans" ADD CONSTRAINT "member.plans_imageID_fkey" FOREIGN KEY ("imageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "peerProfiles" ADD CONSTRAINT "peerProfiles_callToActionImageID_fkey" FOREIGN KEY ("callToActionImageID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
ALTER TABLE "peerProfiles" ADD CONSTRAINT "peerProfiles_logoID_fkey" FOREIGN KEY ("logoID") REFERENCES "images"("id") ON DELETE CASCADE ON UPDATE CASCADE;
```

**Review by Itrulia — APPROVED** [review]

File: packages/api/prisma/migrations/20221026074225_articles_images_authors_guid/migration.sql [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ALTER TABLE "images" ADD COLUMN     "origin" TEXT,
+ALTER COLUMN "id" SET DEFAULT gen_random_uuid();
+
+-- UPDATE IDs on articles table to guid
+ALTER TABLE "articles" ADD "oldId" TEXT NULL;
+UPDATE "articles" SET "oldId" = "id";
+
+UPDATE "articles" SET "id" = gen_random_uuid();
+
+ALTER TABLE "navigations.links" DROP CONSTRAINT "navigations.links_articleID_fkey";
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply packages/api/prisma/migrations/20221026074225_articles_images_authors_guid/migration.sql:10]: Note that this was now merged and has to be adjusted

**Review by Itrulia — CHANGES_REQUESTED** [review]

**Comment by Itrulia** [comment]
As we talked today in the review meeting:

Teasers on pages and articles also have to be migrated.
Article API should check for old and new id.
